include/aws/common/private/system_info_priv.h

```diff
@@ -0,0 +1,36 @@
+#ifndef AWS_COMMON_PRIVATE_SYSTEM_INFO_PRIV_H
+#define AWS_COMMON_PRIVATE_SYSTEM_INFO_PRIV_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/string.h>
+#include <aws/common/system_info.h>
+
+struct aws_system_environment {
+    struct aws_allocator *allocator;
+    struct aws_byte_buf virtualization_vendor;
+    struct aws_byte_buf product_name;
+    enum aws_platform_os os;
+    size_t cpu_count;
+    size_t cpu_group_count;
+    void *additional_impl_data;
```

Comment:
trivial: impl is fine
```suggestion
    void *impl;
```